pym/bob/stringparser.py

```diff
@@ -196,13 +198,169 @@ def getCommand(self):
+            self.op = lambda l, r: l >= r
+        elif op == '==':
+            self.op = lambda l, r: l == r
+        elif op == '!=':
+            self.op = lambda l, r: l != r
+        else:
+            assert False, op
+
+    def __repr__(self):
+        return "BinaryStrOperator({}, {}, {})".format(self.left,
+            self.opStr, self.right)
+
+    def evalExpression(self, env):
+        return self.op(self.left.evalExpressionToString(env), self.right.evalExpressionToString(env))
+
+class IfExpressionParser:
+    def __init__(self):
+
+        # create parsing grammer
+        locationPath = pyparsing.Forward()
+
+        sQStringLiteral = pyparsing.QuotedString("'")
+        sQStringLiteral.setParseAction(
+            lambda s, loc, toks: StringLiteral(s, loc, toks, False))
+
+        dQStringLiteral = pyparsing.QuotedString('"', '\\')
+        dQStringLiteral.setParseAction(
+            lambda s, loc, toks: StringLiteral(s, loc, toks, True))
+
+        stringLiteral = sQStringLiteral | dQStringLiteral
+
+        functionCall = pyparsing.Forward()
+        functionArg = stringLiteral | functionCall
+        functionCall << pyparsing.Word(pyparsing.alphas, pyparsing.alphanums+'-') + \
+            pyparsing.Suppress('(') + \
+            pyparsing.Optional(functionArg +
+                pyparsing.ZeroOrMore(pyparsing.Suppress(',') + functionArg)) + \
+            pyparsing.Suppress(')')
+        functionCall.setParseAction(
+            lambda s, loc, toks: FunctionCall(s, loc, toks))
+
+        predExpr = pyparsing.infixNotation(
+            stringLiteral ^ functionCall ,
+            [
+                ('!',  1, pyparsing.opAssoc.RIGHT, lambda s, loc, toks: NotOperator(s, loc, toks)),
+                ('<',  2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryStrOperator)),
+                ('<=', 2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryStrOperator)),
+                ('>',  2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryStrOperator)),
+                ('>=', 2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryStrOperator)),
+                ('==', 2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryStrOperator)),
+                ('!=', 2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryStrOperator)),
+                ('&&', 2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryBoolOperator)),
+                ('||', 2, pyparsing.opAssoc.LEFT,  infixBinaryOp(BinaryBoolOperator))
+            ])
+
+        self.__ifgrammer = predExpr
+
+    def evalExpression(self, expression, env):
+        try:
+            ret = self.__ifgrammer.parseString(expression)
+        except pyparsing.ParseBaseException as e:
+            raise ParseError("Invalid syntax: " + str(e))
+        return ret[0].evalExpression(env)
+
 
 class Env(MutableMapping):
     def __init__(self, other={}):
         self.data = dict(other)
         self.funs = []
         self.funArgs = {}
         self.touched = [ set() ]
+        self.__expressionParser = IfExpressionParser()
```

Comment:
Env objects are created quite often. AFAICS we only ever need one instance of the `IfExpressionParser`, right? So I think we can lazily create one global instance and use it in `evaluate()` when needed.